client/app/telecom/telephony/alias/configuration/agents/ovhPabx/telecom-telephony-alias-configuration-agents-ovhPabx.controller.js

```diff
@@ -1,170 +1,205 @@
-          this.agents.ids = ids;
-        });
-      })
-      .catch((err) => {
-        this.TucToast.error([this.$translate.instant('telephony_alias_configuration_agents_delete_error'), _.get(err, 'data.message')].join(' '));
-        return this.$q.reject(err);
-      })
-      .finally(() => {
-        this.agents.isDeleting = false;
-        this.agents.selected = {};
+    $onInit() {
+      this.orderedAscAgents = true;
+      this.agents = {
+        ids: [],
+        paginated: [],
+        selected: {},
+        isLoading: false,
+      };
+
+      this.addAgentForm = {
+        numbers: [null],
+        isAdding: false,
+      };
+
+      this.types = [
+        { value: 'internal', label: this.$translate.instant('telephony_alias_configuration_agents_type_internal') },
+        { value: 'external', label: this.$translate.instant('telephony_alias_configuration_agents_type_external') },
+      ];
+
+      this.onChooseServicePopover = ({ serviceName }, pos) => {
+        this.addAgentForm.numbers[pos] = serviceName;
+      };
+
+      this.$q.all({
+        translations: this.$translate.refresh(),
+        services: this.TelephonyMediator.getAll(),
+      }).then((result) => {
+        this.groupList = _.chain(result.services)
+          .pluck('billingAccount')
+          .uniq()
+          .filter(id => _.indexOf(this.hiddenGroups, id) < 0)
+          .map(id => _.find(this.TelephonyMediator.groups, { billingAccount: id }))
+          .value();
       });
-  }
-
-  static startEdition(agent) {
-    _.set(agent, 'inEdition', _.pick(agent, ['description', 'number', 'simultaneousLines', 'status', 'timeout', 'type', 'wrapUpTime']));
-  }
-
-  static isValidAgent({
-    number, timeout, simultaneousLines, wrapUpTime,
-  }) {
-    let valid = true;
-    valid = valid && number;
-    valid = valid && /^\d{1,6}$/.test(timeout);
-    valid = valid && /^\d{1,6}$/.test(wrapUpTime);
-    valid = valid && /^([1-9]|10)$/.test(simultaneousLines);
-    return valid;
-  }
-
-  updateAgent(agent) {
-    _.set(agent, 'isUpdating', true);
-    return this.OvhApiTelephony.OvhPabx().Hunting().Agent().v6()
-      .change({
-        billingAccount: this.$stateParams.billingAccount,
-        serviceName: this.$stateParams.serviceName,
-        agentId: agent.agentId,
-      }, agent.inEdition).$promise
-      .then(() => {
-        _.assign(agent, agent.inEdition);
-        _.set(agent, 'inEdition', null);
-        this.TucToast.success(this.$translate.instant('telephony_alias_configuration_agents_update_success'));
-      })
-      .catch((err) => {
-        this.TucToast.error([this.$translate.instant('telephony_alias_configuration_agents_update_error'), _.get(err, 'data.message')].join(' '));
+
+      return this.fetchAgentsIds().then((ids) => {
+        this.agents.ids = ids;
+      }).catch((err) => {
+        this.TucToast.error([this.$translate.instant('telephony_alias_configuration_agents_get_error'), _.get(err, 'data.message')].join(' '));
```

Comment:
You can use template literal :
```suggestion
        this.TucToast.error(`${this.$translate.instant('telephony_alias_configuration_agents_get_error')} ${_.get(err, 'data.message', ''}`);
```